tla-bmcmt/src/test/scala/at/forsyte/apalache/tla/bmcmt/CrossTestEncodings.scala

```diff
@@ -335,15 +341,54 @@ trait CrossTestEncodings extends AnyFunSuite with Checkers {
     }
   }
 
-  // Ignore this test until the bug described in https://github.com/informalsystems/apalache/issues/1660 is fixed.
+  @nowarn("cat=deprecation&msg=Stream") // ScalaCheck's Shrink still uses Stream, which is deprecated since Scala 2.13.0
+  private def shrinkType(tp: TlaType1): Stream[TlaType1] = tp match {
+    case IntT1() | RealT1() | BoolT1() | StrT1() => Stream.empty
+    case FunT1(arg, res)                         => Stream(arg, res)
+    case SetT1(elem)                             => Stream(elem)
+    case SeqT1(elem)                             => Stream(elem)
+    case TupT1(elems @ _*) =>
+      Stream.concat(
+          // shrink to one of the element types
+          Stream(elems: _*),
+          // recursively shrink one tuple element type
+          elems.zipWithIndex.foldLeft(Stream.empty[TlaType1]) { case (acc, (tp, idx)) =>
+            acc ++ shrink(tp).map(shrunkType => TupT1(elems.updated(idx, shrunkType): _*))
+          },
+          // tuples over a subset of the current element types
+          Stream(shrink(elems).filterNot(_.isEmpty).map(elems => TupT1(elems: _*)): _*),
+      )
+    case RecT1(fieldTypes) =>
+      Stream.concat(
+          // shrink to one of the field types
+          Stream(fieldTypes.values.toSeq: _*),
+          // recursively shrink one record field type
```

Comment:
My suggestion about the order of shrinks of individual elements and the rephrasing of the comment would apply here too.